### PR TITLE
Review Azure

### LINKED ISSUE #2533: Review azure instructions
Several PRs were landed that appear to affect the azure provider, including autoload credentials. Just review this page to make sure.

As per https://github.com/juju/juju/pull/7457
As per https://github.com/juju/juju/pull/7476
As per https://github.com/juju/juju/pull/7477

### PR DESCRIPTION
**Pull request by pmatulis**
resolves #2533 

I started to add some consistency between cloud pages. Apologies. I created issue #2687 to deal specifically with this aspect.

### CODE REVIEW

File: src/en/help-azure-advanced.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+ - application-password
+ - application-name
+ - tenant-id
+
+In the sections below, we will assign each of these a variable name.  When you
+enter them into the command, replace the variable name we give with the actual
+ID that corresponds to the variable.
+
+!!! Important:
+    This process requires the Azure CLI tool to be installed and a successful
```
Comment (degville) [inline src/en/help-azure-advanced.md]: *and a successful login with it* sounds a little too minimal to me (ie. to be installed and used to successfully login to Azure prior to Juju configuration).

File: src/en/help-azure-advanced.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+with an arbitrary application name (typically an internet domain). In our
+example:
+
+```bash
+APP_PASSWORD=some_password
+APP_NAME=ubuntu.example.com
+```
+
+Now create an Active Directory (Kerberos) server principal and grant the
+required resource permissions by assigning a role of ***Owner***:
```
Comment (degville) [inline src/en/help-azure-advanced.md]: It's not exactly clear to me what's being assigned the *Owner* role.

File: src/en/help-azure-advanced.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+  "displayName": "ubuntu.example.com",
+  "name": "http://ubuntu.example.com",
+  "password": "some_password",
+  "tenant": "0fb95fd9-f42f-4c78-94c9-e3d01c2bc5af"
+}
+```
+
+#### `application-id` and `tenant-id`
+
+In the previous output we'll be using the value that follows **appId** as
```
Comment (degville) [inline src/en/help-azure-advanced.md]: *In* doesn't sound right to me, maybe *From*?

File: src/en/help-azure.md [hunk 4]
```diff
@@ -20,151 +21,76 @@ running:
...
-```no-highlight
-Credentials added for cloud azure.
+curl -L https://aka.ms/InstallAzureCli | bash
 ```
 
-If you want to check that the credentials were successfully added, use the
-`juju credentials` command. You will see your Azure credentials listed.
+!!! Note:
+    For instructions that cover installing Azure CLI on Microsoft Windows and
+    Apple macOS, see Microsoft's [Install Azure CLI 2.0][azuretwoinstall]
```
Comment (degville) [inline src/en/help-azure.md]: macOS?

File: src/en/help-azure.md [hunk 5]
```diff
@@ -183,127 +109,80 @@ az account list
...
-required resource permissions by assigning a role of ***Owner***:
+The default choice is `interactive` and it is the recommended method. It is far
+quicker and easier than the manual `service-principal-secret` method. Here,
+we'll assume that 'interactive' has been chosen.
 
-```bash
-az ad sp create-for-rbac --name "ubuntu.example.com" --password $APP_PASSWORD --role Owner
-```
+!!! Note:
+    For guidance on the manul method, see
```
Comment (degville) [inline src/en/help-azure.md]: typo, manual

File: src/en/charms-ha.md [hunk 6]
```diff
@@ -64,8 +65,9 @@ disabled only when bootstrapping the cloud by adding
 juju bootstrap --config availability-sets-enabled=false azure mycloud
 ```
 !!! Note: 
-    By disabling availability sets, you will lose Azure's SLA
-    guarantees. 
+    By disabling availability sets, you will lose Azure's SLA guarantees.
+    [Azure SLA][azure-sla] to learn how availability sets affect uptime
```
[the remark this answers is not in the document]
Comment (pmatulis) [reply src/en/charms-ha.md]: Thanks.

File: src/en/help-azure-advanced.md (hunk 2 above)
Comment (pmatulis) [reply src/en/help-azure-advanced.md]: I agree this sounds convoluted. I think I grabbed this from the Microsoft site when I first wrote it. I'll dig a bit.

File: src/en/help-azure-advanced.md (hunk 3 above)
Comment (pmatulis) [reply src/en/help-azure-advanced.md]: :ok:

File: src/en/help-azure.md (hunk 4 above)
Comment (pmatulis) [reply src/en/help-azure.md]: Should it be something else? I've never used Apple products.

File: src/en/help-azure.md (hunk 5 above)
Comment (pmatulis) [reply src/en/help-azure.md]: Thanks.

File: src/en/help-azure.md (hunk 4 above)
Comment (degville) [reply src/en/help-azure.md]: damn mine eyes. I was sure this was lowercase when I commented! Sorry. And yeah, macOS is the current Apple preference I think.

Comment (pmatulis) [reply src/en/help-azure.md]: :+1:

**Pull request merged**